src/main/java/teammates/ui/controller/InstructorCourseInstructorEditSaveAction.java

```diff
@@ -28,11 +27,29 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
         gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId),
                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
-
-        InstructorAttributes instructorToEdit =
-                extractUpdatedInstructor(courseId, instructorId, instructorName, instructorEmail);
+        String instructorId = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_ID);
+        // get instructor without making any edits
+        InstructorAttributes instructorToEdit;
+        if (instructorId == null) {
+            instructorToEdit = logic.getInstructorForEmail(courseId, instructorEmail);
+        } else {
+            instructorToEdit = logic.getInstructorForGoogleId(courseId, instructorId);
+        }
+        boolean isDisplayedToStudents = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT) != null;
+        int numOfInstrDisplayed = getNumberOfInstructorsDisplayedToStudents(courseId, instructorToEdit.isDisplayedToStudents,
+                                                                             isDisplayedToStudents);
+        /* This if statement should only be processed if only one instructor is displayed to students and you are
```

Comment:
Try leaving a blank line between logical blocks of code (eg. here, in between getting the instructor, and checking for the number of instructors still being displayed). This helps developers to more easily read the code.